plone/app/contenttypes/migration/field_migrators.py

```diff
@@ -17,17 +17,24 @@ def migrate_simplefield(src_obj, dst_obj, src_fieldname, dst_fieldname):
     to the target-object. The only transform is a safe_unicode of the value.
     """
     field = src_obj.getField(src_fieldname)
-    if field:
-        at_value = field.get(src_obj)
+    try:
+        if field:
+            # Check for existence first
+            field.getStorage(src_obj).get(field.getName(), src_obj)
+            at_value = field.get(src_obj)
+        else:
+            at_value = getattr(src_obj, src_fieldname)
+            if at_value and hasattr(at_value, '__call__'):
+                at_value = at_value()
+    except AttributeError:
+        # The value is "unset" on the old instance and so should be unset on
+        # the new instance
+        pass
     else:
-        at_value = getattr(src_obj, src_fieldname, None)
-        if at_value and hasattr(at_value, '__call__'):
-            at_value = at_value()
-    if isinstance(at_value, tuple):
-        at_value = tuple(safe_unicode(i) for i in at_value)
-    if isinstance(at_value, list):
-        at_value = [safe_unicode(i) for i in at_value]
-    if at_value:
```

Comment:
This line is still needed, right?

Reply:
That's several lines, I'm not clear what you're asking?

Reply:
The line `if at_value:` was removed, it seems to me like it is needed.

Reply:
No, that's the actual point where the bug happens.  See the commit message: 

> If an AT schema had a `BooleanField` that defaults to `True` and an instance
> had a value set to `False`, the migration would take that `False` to mean to
> leave the value as unset on the new DX instance which would then get the
> default `True`

Reply:
Oh! You are right, sorry for that!